ci/init.js

```diff
@@ -14,6 +25,11 @@ const options = {
 
 let shouldInit = true
 
+if (isPackageManager()) {
```

Comment:
@juan-fernandez Why do this only for CI? We pretty much never want the library as a whole to instrument package managers (or anything else in the denylist).

Reply:
> does APM or other products suggest users to init dd-trace via NODE_OPTIONS='-r dd-trace/init'?

Yes, the recommended initialization is with `NODE_OPTIONS`, and other approaches are discouraged (and will also be removed when the new public API lands). There are also some environments where programmatic init cannot be done and `NODE_OPTIONS` must be used.

> If the initialization is programmatic (as suggested in the docs:

Not sure why this is still the case today as we've been recommending `NODE_OPTIONS` for years now 😅